src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/IServiceProviderExtensions.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS
+{
+    internal static class IServiceProviderExtensions
+    {
+        /// <summary>
+        /// Returns the specified interface from the service. This is useful when the service and interface differ
+        /// </summary>
+        public static InterfaceType GetService<InterfaceType, ServiceType>(this IServiceProvider sp)
+            where InterfaceType : class
+            where ServiceType : class
+        {
+            InterfaceType service = null;
+
+            try
+            {
+                service = sp.GetService(typeof(ServiceType)) as InterfaceType;
+            }
+            catch
+            {
```

Comment:
Change you change this to 1) not use an "as" cast and 2) not to catch exceptions. Both of these hide bugs.